pkg/controller/config-controller.go

```diff
@@ -0,0 +1,313 @@
+				// Two different versions of the same PVCs will always have different RVs.
+				return
+			}
+			c.handleObjUpdate(new)
+		},
+
+		DeleteFunc: c.handleObjDelete,
+	})
+
+	return c
+}
+
+func (c *ConfigController) handleObjAdd(obj interface{}) {
+	c.handleObject(obj, "add")
+}
+func (c *ConfigController) handleObjUpdate(obj interface{}) {
+	c.handleObject(obj, "update")
+}
+func (c *ConfigController) handleObjDelete(obj interface{}) {
+	c.handleObject(obj, "delete")
+}
+
+func (c *ConfigController) handleObject(obj interface{}, verb string) {
+	var object metav1.Object
+	var ok bool
+	if object, ok = obj.(metav1.Object); !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			runtime.HandleError(errors.Errorf("error decoding object, invalid type"))
+			return
+		}
+		object, ok = tombstone.Obj.(metav1.Object)
+		if !ok {
+			runtime.HandleError(errors.Errorf("error decoding object tombstone, invalid type"))
+			return
+		}
+		glog.V(3).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
+	}
+	glog.V(3).Infof("Processing object: %s", object.GetName())
+
+	var url string
+	if ing, ok := obj.(*extensionsv1beta1.Ingress); ok {
+		url = getURLFromIngress(ing, c.uploadProxyServiceName)
+
+	}
+	if route, ok := obj.(*routev1.Route); ok {
+		url = getURLFromRoute(route, c.uploadProxyServiceName)
+	}
+
+	config, err := c.cdiClientSet.CdiV1alpha1().CDIConfigs(util.GetNamespace()).Get("config", metav1.GetOptions{})
+	if err != nil {
+		runtime.HandleError(errors.Errorf("error getting CDI config: %s", err))
+		return
+	}
+	if verb == "delete" {
+		if *config.Spec.Status.UploadProxyURL == url {
+			url = ""
+		}
+	}
+	err = updateCDIConfig(c.cdiClientSet, config, url)
+	if err != nil {
+		runtime.HandleError(errors.Errorf("error updating CDI config: %s", err))
+		return
+	}
+	return
+
+}
+
+func (c *ConfigController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *ConfigController) processNextWorkItem() bool {
+	obj, shutdown := c.queue.Get()
+
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer c.queue.Done(obj)
+
+		var key string
+		var ok bool
+
+		if key, ok = obj.(string); !ok {
+			c.queue.Forget(obj)
+			runtime.HandleError(errors.Errorf("expected string in workqueue but got %#v", obj))
+			return nil
+		}
+
+		if err := c.syncHandler(key); err != nil {
+			return errors.Errorf("error syncing '%s': %s", key, err.Error())
+		}
+
+		c.queue.Forget(obj)
+		glog.Infof("Successfully synced '%s'", key)
+		return nil
+
+	}(obj)
+
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+func (c *ConfigController) syncHandler(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(errors.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the CDIConfig resource with this namespace/name
+	config, err := c.configLister.CDIConfigs(namespace).Get(name)
```

Comment:
You have to figure out what object(s) you will be queueing.  Is it the CDIConfig or or the Ingress/Routes?

If you queue the CDIConfig, you fill have to go through all the routes/ingresses here.  If you queue the ingress/routes, I think you have to have multiple handler functions/queues.